src/analysis/functions.rs

```diff
@@ -206,14 +210,32 @@ fn analyze_function(
     configured_functions: &[&config::functions::Function],
     imports: &mut Imports,
 ) -> Info {
-    let async = func.parameters.iter().any(|parameter| parameter.scope == ParameterScope::Async);
+    let async = func.name.ends_with("_async") ||
+                func.parameters.iter()
+                               .any(|parameter| parameter.scope == ParameterScope::Async &&
+                                                parameter.c_type == "GAsyncReadyCallback");
+    let expecting_data = !async &&
```

Comment:
This could probably get a nicer name. `has_callback_parameter`, i.e. the function has a callback parameter?

Reply:
Indeed, thanks for coming up with a better naming (I'm so bad at that ><)!